test/parallel/test-crypto-keygen.js

```diff
@@ -1543,44 +1543,65 @@ const sec1EncExp = (cipher) => getRegExpForPEM('EC PRIVATE KEY', cipher);
   }
 }
 
-if (!common.hasOpenSSL3) {
-  // Passing an empty passphrase string should not cause OpenSSL's default
-  // passphrase prompt in the terminal.
-  // See https://github.com/nodejs/node/issues/35898.
-
-  for (const type of ['pkcs1', 'pkcs8']) {
-    generateKeyPair('rsa', {
-      modulusLength: 1024,
-      privateKeyEncoding: {
-        type,
-        format: 'pem',
-        cipher: 'aes-256-cbc',
-        passphrase: ''
-      }
-    }, common.mustSucceed((publicKey, privateKey) => {
-      assert.strictEqual(publicKey.type, 'public');
+// Passing an empty passphrase string should not cause OpenSSL's default
+// passphrase prompt in the terminal.
+// See https://github.com/nodejs/node/issues/35898.
 
-      for (const passphrase of ['', Buffer.alloc(0)]) {
-        const privateKeyObject = createPrivateKey({
-          passphrase,
-          key: privateKey
-        });
-        assert.strictEqual(privateKeyObject.asymmetricKeyType, 'rsa');
-      }
+for (const type of ['pkcs1', 'pkcs8']) {
+  generateKeyPair('rsa', {
+    modulusLength: 1024,
+    privateKeyEncoding: {
+      type,
+      format: 'pem',
+      cipher: 'aes-256-cbc',
+      passphrase: ''
+    }
+  }, common.mustSucceed((publicKey, privateKey) => {
+    assert.strictEqual(publicKey.type, 'public');
 
-      // Encrypting with an empty passphrase is not the same as not encrypting
-      // the key, and not specifying a passphrase should fail when decoding it.
-      assert.throws(() => {
-        return testSignVerify(publicKey, privateKey);
-      }, {
-        name: 'TypeError',
-        code: 'ERR_MISSING_PASSPHRASE',
-        message: 'Passphrase required for encrypted key'
+    for (const passphrase of ['', Buffer.alloc(0)]) {
+      const privateKeyObject = createPrivateKey({
+        passphrase,
+        key: privateKey
       });
-    }));
-  }
+      assert.strictEqual(privateKeyObject.asymmetricKeyType, 'rsa');
+    }
+
+    // Encrypting with an empty passphrase is not the same as not encrypting
+    // the key, and not specifying a passphrase should fail when decoding it.
+    assert.throws(() => {
+      return testSignVerify(publicKey, privateKey);
+    }, common.hasOpenSSL3 ? {
+      name: 'Error',
+      code: 'ERR_OSSL_CRYPTO_INTERRUPTED_OR_CANCELLED',
```

Comment:
The 1.1.1 error message certainly looks more helpful here. Probably not blocking on this, but maybe we should fix that.

Reply:
@tniessen here's my attempt to fix this - https://github.com/nodejs/node/pull/42400. PTAL. :)